lib/Sema/DerivedConformanceCodingKey.cpp

```diff
@@ -405,8 +405,9 @@ static bool canSynthesizeCodingKey(TypeChecker &tc, Decl *parentDecl,
       return false;
   }
 
-  if (!enumDecl->getInherited().empty() &&
-      enumDecl->getInherited().front().isError())
+  auto inherited = enumDecl->getInherited();
+  if (!inherited.empty() && inherited.front().wasValidated() &&
```

Comment:
Either it's reasonable to validate the enum here to make sure its raw type is resolved, or it's reasonable to *require* that the enum has already been validated on all paths that can get here. Bailing out early here would be incorrect because it could lead to something not being synthesized in time for a lookup or conformance check.